Beta: issues with types

The types were previously imported fine, but with the beta something has broken. Now I get:
`.storybook/preview.ts(10,42): error TS2307: Cannot find module 'msw-storybook-addon' or its corresponding type declarations.`

Happy to debug somehow if I can.

Thanks @kettanaito!
I updated 

<details>

<summary><code>node_modules/msw-storybook-addon/package.json</code></summary>

```json
{
  "name": "msw-storybook-addon",
  "description": "Mock API requests in Storybook with Mock Service Worker.",
  "version": "2.0.0-beta.1",
  "exports": {
    ".": {
      "browser": {
        "types": "./dist/index.browser.d.ts",
        "default": "./dist/index.browser.js"
      },
      "react-native": {
        "types": "./dist/index.react-native.d.ts",
        "default": "./dist/index.react-native.js"
      },
      "node": {
        "types": "./dist/index.node.d.ts",
        "default": "./dist/index.node.js"
      }
    },
    "./package.json": "./package.json"
  },
  "scripts": {
    "dev": "yarn build --watch",
    "clean": "rimraf ./dist",
    "build": "tsup",
    "prepublishOnly": "yarn clean && yarn build",
    "release": "auto shipit"
  },
  "files": [
    "dist/"
  ],
  "repository": {
    "type": "git",
    "url": "git+https://github.com/mswjs/msw-storybook-addon.git",
    "directory": "packages/msw-addon"
  },
  "keywords": [
    "storybook-addon",
    "msw",
    "mocking",
    "api",
    "graphql",
    "network",
    "data-state"
  ],
  "author": "Aditya Agarwal (@dev__adi)",
  "bugs": {
    "url": "https://github.com/mswjs/msw-storybook-addon/issues"
  },
  "homepage": "https://msw-sb.vercel.app/",
  "license": "MIT",
  "dependencies": {
    "is-node-process": "^1.0.1"
  },
  "devDependencies": {
    "@auto-it/released": "^10.32.2",
    "auto": "^10.32.2",
    "msw": "^2.0.9",
    "rimraf": "^5.0.5",
    "tsup": "^8.0.1",
    "typescript": "^5.2.2"
  },
  "peerDependencies": {
    "msw": "^2.0.0"
  },
  "storybook": {
    "displayName": "Mock Service Worker",
    "icon": "https://user-images.githubusercontent.com/1671563/144888802-84346d8f-77c9-4377-98c7-4b0364797978.png"
  },
  "auto": {
    "prereleaseBranches": [
      "next",
      "prerelease"
    ],
    "plugins": [
      "npm",
      "released"
    ]
  }
}
```

</details>

Then ran `tsc` again in my project folder.
I get the error:

```
$ npx tsc
.storybook/preview.ts:2:53 - error TS2307: Cannot find module 'msw-storybook-addon' or its corresponding type declarations.

2 import { initialize, mswLoader, mswDecorator } from "msw-storybook-addon";
                                                      ~~~~~~~~~~~~~~~~~~~~~
```

Is there an additional step needed to rebuild?

> Thanks @kettanaito! I updated
> 
> `node_modules/msw-storybook-addon/package.json`
> Then ran `tsc` again in my project folder. I get the error:
> 
> ```
> $ npx tsc
> .storybook/preview.ts:2:53 - error TS2307: Cannot find module 'msw-storybook-addon' or its corresponding type declarations.
> 
> 2 import { initialize, mswLoader, mswDecorator } from "msw-storybook-addon";
>                                                       ~~~~~~~~~~~~~~~~~~~~~
> ```
> 
> Is there an additional step needed to rebuild?

Can you try to rerun your TypeScript server? On VSCode for instance is "cmd shift P" then
<img width="291" alt="image" src="https://github.com/mswjs/msw-storybook-addon/assets/1671563/156ca4db-b258-4f2b-92db-fdf8e46bfcf7">

@yannbf I tried in VSCode as well, before and after restarting the VSCode language server.
I still see the same error.

<details>

```
.storybook/preview.ts:2:53 - error TS2307: Cannot find module 'msw-storybook-addon' or its corresponding type declarations.
```

</details>

--- a/packages/msw-addon/package.json
+++ b/packages/msw-addon/package.json
@@ -2,20 +2,16 @@
   "name": "msw-storybook-addon",
   "description": "Mock API requests in Storybook with Mock Service Worker.",
   "version": "1.6.0",
+  "type": "module",
   "main": "./dist/index.browser.js",
   "types": "./dist/index.browser.d.ts",
   "exports": {
-    "browser": {
+    ".": {
       "types": "./dist/index.browser.d.ts",
+      "browser": "./dist/index.browser.js",
+      "react-native": "./dist/index.react-native.js",
+      "node": "./dist/index.node.js",
       "default": "./dist/index.browser.js"
-    },
-    "react-native": {
-      "types": "./dist/index.react-native.d.ts",
-      "default": "./dist/index.react-native.js"
-    },
-    "node": {
-      "types": "./dist/index.node.d.ts",
-      "default": "./dist/index.node.js"
     }
   },
   "scripts": {

--- a/packages/msw-addon/src/applyRequestHandlers.ts
+++ b/packages/msw-addon/src/applyRequestHandlers.ts
@@ -1,6 +1,6 @@
 import type { RequestHandler } from 'msw'
 import { api } from '@build-time/initialize'
-import type { Context } from './decorator'
+import type { Context } from './decorator.js'
 
 export function applyRequestHandlers(
   handlersListOrObject: Context['parameters']['msw']

--- a/packages/msw-addon/src/augmentInitializeOptions.ts
+++ b/packages/msw-addon/src/augmentInitializeOptions.ts
@@ -1,4 +1,4 @@
-import { InitializeOptions } from "./initialize";
+import { InitializeOptions } from "./initialize.js";
 
 const fileExtensionPattern = /\.(js|jsx|ts|tsx|mjs|woff|woff2|ttf|otf|eot)$/;
 const filteredURLSubstrings = [

--- a/packages/msw-addon/src/decorator.ts
+++ b/packages/msw-addon/src/decorator.ts
@@ -1,5 +1,5 @@
 import type { RequestHandler } from 'msw'
-import { applyRequestHandlers } from './applyRequestHandlers'
+import { applyRequestHandlers } from './applyRequestHandlers.js'
 
 export type MswParameters = {
   msw?:

--- a/packages/msw-addon/src/index.ts
+++ b/packages/msw-addon/src/index.ts
@@ -1,3 +1,3 @@
 export { initialize, InitializeOptions, getWorker } from '@build-time/initialize'
-export * from './decorator'
-export * from './loader'
+export * from './decorator.js'
+export * from './loader.js'

--- a/packages/msw-addon/src/initialize.browser.ts
+++ b/packages/msw-addon/src/initialize.browser.ts
@@ -1,7 +1,7 @@
 import type { RequestHandler } from 'msw'
 import type { SetupWorker } from 'msw/browser'
 import { setupWorker } from 'msw/browser'
-import { augmentInitializeOptions } from './augmentInitializeOptions'
+import { augmentInitializeOptions } from './augmentInitializeOptions.js'
 
 export let api: SetupWorker
 

--- a/packages/msw-addon/src/initialize.node.ts
+++ b/packages/msw-addon/src/initialize.node.ts
@@ -1,7 +1,7 @@
 import type { RequestHandler } from 'msw'
 import type { SetupServer } from 'msw/node'
 import { setupServer } from 'msw/node'
-import { augmentInitializeOptions } from './augmentInitializeOptions'
+import { augmentInitializeOptions } from './augmentInitializeOptions.js'
 
 export let api: SetupServer
 

--- a/packages/msw-addon/src/initialize.react-native.ts
+++ b/packages/msw-addon/src/initialize.react-native.ts
@@ -1,7 +1,7 @@
 import type { RequestHandler } from 'msw'
 import type { SetupServer } from 'msw/node'
 import { setupServer } from 'msw/native'
-import { augmentInitializeOptions } from './augmentInitializeOptions'
+import { augmentInitializeOptions } from './augmentInitializeOptions.js'
 
 export let api: SetupServer
 

--- a/packages/msw-addon/src/loader.ts
+++ b/packages/msw-addon/src/loader.ts
@@ -1,5 +1,5 @@
-import type { Context } from './decorator'
-import { applyRequestHandlers } from './applyRequestHandlers'
+import type { Context } from './decorator.js'
+import { applyRequestHandlers } from './applyRequestHandlers.js'
 
 export const mswLoader = async (context: Context) => {
   applyRequestHandlers(context.parameters.msw)

--- a/packages/msw-addon/tsconfig.json
+++ b/packages/msw-addon/tsconfig.json
@@ -3,12 +3,9 @@
     "strict": true,
     "outDir": "./dist",
     "noImplicitAny": true,
-    "module": "commonjs",
+    "module": "Node16",
     "target": "es5",
-    "skipLibCheck": true,
     "allowJs": true,
-    "moduleResolution": "node",
-    "allowSyntheticDefaultImports": true,
     "declaration": true,
     "baseUrl": ".",
     "paths": {

--- a/packages/msw-addon/tsup.config.ts
+++ b/packages/msw-addon/tsup.config.ts
@@ -6,7 +6,7 @@ const browser = defineConfig({
   },
   outDir: './dist',
   target: ['chrome112'],
-  format: 'cjs',
+  format: 'esm',
   esbuildOptions(options) {
     options.alias = {
       ...(options.alias || {}),
@@ -24,7 +24,7 @@ const node = defineConfig({
   },
   outDir: './dist',
   target: 'node18',
-  format: 'cjs',
+  format: 'esm',
   esbuildOptions(options) {
     options.alias = {
       ...(options.alias || {}),
@@ -42,7 +42,7 @@ const reactNative = defineConfig({
   },
   outDir: './dist',
   target: 'esnext',
-  format: 'cjs',
+  format: 'esm',
   esbuildOptions(options) {
     options.alias = {
       ...(options.alias || {}),

